lxd/images.go

```diff
@@ -60,17 +61,38 @@ func detectCompression(fname string) ([]string, string, error) {
 
 }
 
+var canMknod = true
+
+func checkCanMknod() {
+	/* TODO - mktemp */
+	fnam := shared.VarPath("null")
+	// warning to cut-pasters: can't do the below in general, that is if minor is big
+	if err := syscall.Mknod(fnam, syscall.S_IFCHR, int((int64(1)<<8)|int64(3))); err != nil {
+		canMknod = false
+		os.Remove(fnam)
+	}
+}
+
 func untar(tarball string, path string) error {
 	extractArgs, _, err := detectCompression(tarball)
 	if err != nil {
 		return err
 	}
 
-	args := []string{"-C", path, "--numeric-owner"}
+	command := "tar"
+	args := []string{}
+	if !canMknod {
+		// if we are running in a userns where we cannot mknod,
+		// then run with a seccomp filter which turns mknod into a
```

Comment:
I think this comment is wrong, you're not using a seccomp filter per your commit log.